File: backend/src/main/java/bg/sofia/uni/fmi/tbb/model/Route.java
package bg.sofia.uni.fmi.tbb.model;

import lombok.*;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.util.List;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
@NoArgsConstructor
public class Route {

    @NotNull
    @NonNull
    private Stop startPoint;

    @NotNull
    @NonNull
    private Stop endPoint;

    @NotNull
    @NonNull
    @Min(0)
    private double distance;

    @NotNull
    @NonNull
    @Min(0)
    private double duration;

    private List<Stop> intermediateStops;

}

File: frontend/src/app/_models/Route.ts
import { Stop } from "./Stop";

export class Route {
    id?: string;
    startPoint: Stop;
    endPoint: Stop;
    duration: number;
    distance: number;

    constructor(startPoint: Stop, endPoint: Stop, duration: number, distance: number) {
        this.startPoint = startPoint;
        this.endPoint = endPoint;
        this.duration = duration;
        this.distance = distance;
    }
}